helm/charts/determined/scripts/k8s-password-change.py

```diff
@@ -55,35 +55,40 @@ def getMasterAddress(
     target_service = f"determined-master-service-{service_name}"
 
     if node_port != "true":
-        while True:
+        n=300 # 5 minutes
+        while n > 0: 
```

Comment:
It is not the same, I guess, because the cycle just loop, and inside there already a for loop with the break. 

The original code that you can see here, is quite involuted:

```
        **while True:**
            services = requests.get(
                url=f"https://{service_host}:{service_port}/api/v1/namespaces/{namespace}/services",
                headers={"Authorization": f"Bearer {token}"},
                verify=False,
            ).json()
            
            for svc in services["items"]:
                if target_service in svc["metadata"]["name"]:
                    status = svc["status"]["loadBalancer"]
                    if "ingress" not in svc["status"]["loadBalancer"] or status["ingress"] is None:

                        time.sleep(1)
                        break

                    if status["ingress"][0].get("hostname"):
                        # use hostname over ip address, if available
                        return f"{status['ingress'][0]['hostname']}:{master_port}"
                    else:
                        return f"{status['ingress'][0]['ip']}:{master_port}"
```
I've just removed an infinite loop `while true:` changing it with a finite loop that after 5 minutes gives up, but the entire code structure remained the same.

```
   n=300 # 5 minutes
    while n > 0:
            services = requests.get(
                url=f"https://{service_host}:{service_port}/api/v1/namespaces/{namespace}/services",
                headers={"Authorization": f"Bearer {token}"},
                verify=False,
            ).json()

            for svc in services["items"]:
                if target_service in svc["metadata"]["name"]:
                    status = svc["status"]["loadBalancer"]
                    if "ingress" not in svc["status"]["loadBalancer"] or status["ingress"] is None:

                        n = n - 1
                        time.sleep(1) # 1 second and loop
                        break

                    if status["ingress"][0].get("hostname"):
                        # use hostname over ip address, if available
                        return f"{status['ingress'][0]['hostname']}:{master_port}"
                    else:
                        return f"{status['ingress'][0]['ip']}:{master_port}"
```

I did that to avoid an infinite loop that, potentially, in some cases, could stop the deployment.

Reply:
Changed